File: frontend/types/job.ts
export type JobType = "email_sync" | "cfdi_parse" | "batch_zip" | "analytics_refresh" | "report_export";
export type JobStatus = "pending" | "running" | "completed" | "failed" | "cancelled";

export interface ProcessingJob {
  id: string;
  jobType: JobType;
  status: JobStatus;
  progress: number;
  totalItems: number | null;
  processedItems: number;
  failedItems: number;
  outputArtifactId: string | null;
  errorDetail: string | null;
  startedAt: string | null;
  completedAt: string | null;
  createdAt: string;
}

File: backend/app/schemas/job.py
import uuid
from datetime import datetime
from typing import Any

from pydantic import BaseModel


class ProcessingJobOut(BaseModel):
    id: uuid.UUID
    job_type: str
    status: str
    progress: int
    total_items: int | None
    processed_items: int
    failed_items: int
    output_artifact_id: uuid.UUID | None
    error_detail: str | None
    started_at: datetime | None
    completed_at: datetime | None
    created_at: datetime

    model_config = {"from_attributes": True}


class ProcessingJobList(BaseModel):
    items: list[ProcessingJobOut]
    total: int
    page: int
    page_size: int
